core/src/main/java/io/methvin/watcher/DirectoryWatcher.java

```diff
@@ -328,13 +365,73 @@ public FileVisitResult preVisitDirectory(Path dir, BasicFileAttributes attrs) th
+    // If we don't use file tree, we don't register `directory` for recursive file watching
+    Watchable watchable = isMac ? new WatchablePath(directory, useFileTreeModifier) : directory;
     WatchEvent.Modifier[] modifiers = useFileTreeModifier
         ? new WatchEvent.Modifier[] {ExtendedWatchEventModifier.FILE_TREE}
         : new WatchEvent.Modifier[] {};
     WatchEvent.Kind<?>[] kinds = new WatchEvent.Kind<?>[] {ENTRY_CREATE, ENTRY_DELETE, ENTRY_MODIFY};
     WatchKey watchKey = watchable.register(watchService, kinds, modifiers);
     keyRoots.put(watchKey, directory);
+    watchingPathRoots.add(directory);
+  }
+
+  /**
+   * Register parents of independent files for non-recursive event watching.
+   *
+   * If any of the parents of the independent files have already been registered or
+   * are subsumed by a path that has been registered, they are skipped.
+   *
+   * @param files The independent files that we want to watch.
+   */
+  private void registerRootsForFiles(List<Path> files) throws IOException {
+    for (Path file: files) {
+      Path parentFile = file.getParent();
+     if (!watchingPathRoots.contains(parentFile)) {
+        if (watchingPathRoots.isEmpty()) {
+          nonRecursivePathRoots.add(parentFile);
+          register(parentFile, false);
+        } else {
+          for (Path watchRoot : watchingPathRoots) {
+            if (parentFile.startsWith(watchRoot)) {
+              logger.debug("Parent file " + parentFile.toString() + " is contained in " + watchRoot);
+            } else {
+              nonRecursivePathRoots.add(parentFile);
+              register(parentFile, false);
+            }
+          }
+        }
+      } else {
+        logger.debug("Parent file " + parentFile.toString() + " already is being watched");
+      }
+    }
+  }
+
+  /**
+   * Checks whether a directory path associated with an event should be processed or not.
+   * Note that the argument path needs to represent a directory.
+   *
+   * A more efficient implementation could be done by checking directly that the parent
+   * of a path is a parent of a non-recursive directory. However, this implementation is
+   * not viable as the order of the event processing is not ensured.
+   *
+   * For example, take a non-recursive directory `foo`. If a user creates `foo/bar/foo.txt`,
+   * then there's no guarantee that the creation event of the directory `foo/bar` will be
+   * processed before the creation event of `foo/bar/foo.txt`.
+   *
+   * @param directory The direcotry path associated with an event of any kind.
+   * @return Whether directory watcher should process the event or not.
+   */
+  private boolean shouldIgnoreDirectoryEvent(Path directory) {
+    boolean ignoreEvent = false;
+    Path parentPath = directory.getParent();
+    for (Path nonRecursivePath : nonRecursivePathRoots) {
+      if (parentPath.equals(nonRecursivePath) && directory.startsWith(nonRecursivePath)) {
+        ignoreEvent = true;
```

Comment:
Any reason why you used a variable as opposed to `return true` here?

Reply:
Nope, I can do `return true` here instead